### ISSUE TITLE
parser: add warnings to pure annotations which parser is not able to interpret

### ISSUE THREAD

**Issue by h-a-n-a** [oxc-project/oxc#20334]
It's good to have warnings in some scenario where pure annotations are not able 
to be interpreted due to the position of the comment.

For example, pure annotation in the example down below is not able 
to be interpreted due to position and it's ignored by the parser.

```js
/* #__PURE__ */ React.createElement;
```

whereas, rollup would emit this warning to help people adjust the placement of pure annotation.

```
Rollup completed with warnings:
main.js (1:0): A comment "/* #__PURE__ */" in "main.js" contains an annotation that Rollup cannot interpret due to the position of the comment. The comment will be removed to avoid issues. [(link)](https://rollupjs.org/configuration-options/#pure)
/main.js (1:0)
1: /* #__PURE__ */ React.createElement;
```

https://rollupjs.org/repl/?version=4.59.0&shareable=eyJleGFtcGxlIjpudWxsLCJtb2R1bGVzIjpbeyJjb2RlIjoiLyogI19fUFVSRV9fICovIFJlYWN0LmNyZWF0ZUVsZW1lbnQ7IiwiaXNFbnRyeSI6dHJ1ZSwibmFtZSI6Im1haW4uanMifV0sIm9wdGlvbnMiOnt9fQ==

**Comment by Boshen** [comment]
We need a follow up in Rolldown after PR is merged.

**Comment by h-a-n-a** [comment]
I'm happy to help after the PR is merged.
